### PR TITLE
Update Groq models and tokens

### PR DESCRIPTION
**Pull request by dceoy**
### **User description**
- Add new models and remove deprecated models for Groq
  - cf. https://console.groq.com/docs/models

___

### **PR Type**
Enhancement

___

### **Description**
- Updated Groq model configurations with latest available models and correct token limits
- Added new models:
  - `groq/llama-3.3-70b-versatile`
  - `groq/gemma2-9b-it`
- Removed deprecated models:
  - `groq/llama-3.1-70b-versatile`
  - `groq/llama-3.1-405b-reasoning`
- Fixed token limit for `groq/llama-3.1-8b-instant` from 131072 to 8192

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Configuration changes</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>__init__.py</strong><dd><code>Update Groq models configuration and token limits</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pr_agent/algo/__init__.py

<li>Updated Groq model tokens and names<br> <li> Removed deprecated models: <code>llama-3.1-70b-versatile</code> and <br><code>llama-3.1-405b-reasoning</code><br> <li> Added new models: <code>llama-3.3-70b-versatile</code> and <code>gemma2-9b-it</code><br> <li> Fixed token limit for <code>llama-3.1-8b-instant</code> from 131072 to 8192<br>

</details>

  </td>
  <td><a href="https://github.com/Codium-ai/pr-agent/pull/1421/files#diff-5923c546f24ec7308a0e43fc84bb6fe40de7bfe2ac6ee842da9578e5dc2c692b">+3/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

File: pr_agent/algo/__init__.py [hunk 1]
```diff
@@ -66,10 +66,10 @@
     'claude-3-5-sonnet': 100000,
     'groq/llama3-8b-8192': 8192,
     'groq/llama3-70b-8192': 8192,
+    'groq/llama-3.1-8b-instant': 8192,
+    'groq/llama-3.3-70b-versatile': 8192,
```
Comment (mrT23) [inline pr_agent/algo/__init__.py:70]: why 8192 ?

from your link:

image

Comment (mrT23) [reply pr_agent/algo/__init__.py:70]: make sure the tokens are correct. 8192 seems small to a modern model

Comment (dceoy) [reply pr_agent/algo/__init__.py:70]: I retrieved the max tokens from the configuration of LiteLLM:
https://github.com/BerriAI/litellm/blob/main/model_prices_and_context_window.json

Comment (mrT23) [reply pr_agent/algo/__init__.py:70]: "max_tokens" is max_input_tokens

image

Comment (dceoy) [reply pr_agent/algo/__init__.py]: Okay, I updated it!

Comment (mrT23) [reply pr_agent/algo/__init__.py]: please review and fix the tokens for the other models

Comment (dceoy) [reply pr_agent/algo/__init__.py]: I've already reviewed them.

Comment (mrT23) [reply pr_agent/algo/__init__.py]: 'groq/llama-3.1-8b-instant': 8192,

Comment (mrT23) [reply pr_agent/algo/__init__.py]: image

Comment (mrT23) [reply pr_agent/algo/__init__.py]: from the main link you gave

Comment (dceoy) [reply pr_agent/algo/__init__.py]: https://github.com/BerriAI/litellm/blob/24dd6559a68ddf2bbb9642c48c3482d0058b3e75/model_prices_and_context_window.json#L2159-L2161

**Review by mrT23 — APPROVED** [review]

**Pull request merged**